### ISSUE TITLE
1.16.1 - Command SwiftEmitModule failed with a nonzero exit code

### ISSUE THREAD

**Issue by filipealva** [CocoaPods/CocoaPods#12673]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

Run `pod install` 

## What did you expect to happen?

To find my project building without issues

## What happened instead?

`Filestack` pod started returning an error after I updated cocoapods to `1.16.1` to fix an issue on another pod related to a build settings flag

The `Filestack`error: `Command SwiftEmitModule failed with a nonzero exit code`

### Stack

```
   CocoaPods : 1.16.1
        Ruby : ruby 2.6.10p210 (2022-04-12 revision 67958) [arm64-darwin22]
    RubyGems : 3.0.9
        Host : macOS 14.4.1 (23E224)
       Xcode : 15.2 (15C500b)
         Git : git version 2.39.3 (Apple Git-145)
Ruby lib dir : /Users/filipealva/.rvm/rubies/ruby-2.6.10/lib
Repositories : trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /Users/filipealva/.rvm/gems/ruby-2.6.10/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.5
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.1
cocoapods-trunk       : 1.6.0
cocoapods-try         : 1.2.0
```

## Project that demonstrates the issue

N/A

**Comment by filipealva** [comment]
Update here:

the problem was that my pods (and my project) wouldn't build with `xcodeproj` 1.26.0 

enforcing on my gemfile that it shouldn't update `xcodeproj` further than `1.25.0` fixed the issue

also had to downgrade to `1.15.2` but just downgrading wouldn't do it because even on cocoapods `1.15.2` it allows the `xcodeproj` to be up to `2.0` so fastlane would update `xcodeproj` to the latest version available regardless that I was on `1.15.2`


hope it helps more people

**Comment by ng-cen** [comment]
in Podfile, disable Swift module interface verification works for me:

```
config.build_settings['SWIFT_EMIT_MODULE_INTERFACE'] = 'NO'
config.build_settings['SWIFT_COMPILATION_MODE'] = 'wholemodule'
```
code snipet:
```
post_install do |installer|
      react_native_post_install(
        installer,
        config[:reactNativePath],
        # Set `mac_catalyst_enabled` to `true` in order to apply patches
        # necessary for Mac Catalyst builds
        :mac_catalyst_enabled => false
      )
    
      installer.pods_project.targets.each do |target|
        target.build_configurations.each do |config|
          
          # ** IMPORTANT Fix for RN 0.75.5 **
          # Fix xcode build error: Command SwiftVerifyEmittedModuleInterface failed with a nonzero exit code
          # Packages in this array will disable Swift module interface verification
          swift_problematic_targets = [
            'react-native-image-marker',
            'react-native-vision-camera',
            'VisionCamera',
            'react-native-geolocation-service',
            'sparkfabrik-react-native-idfa-aaid',
            'lottie-ios',
            'lottie-react-native'
          ]
          if swift_problematic_targets.include?(target.name)
            config.build_settings['SWIFT_EMIT_MODULE_INTERFACE'] = 'NO'
            config.build_settings['SWIFT_COMPILATION_MODE'] = 'wholemodule'
          end
        end
      end
```
